plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/preferences/PrefPageConfirmations.java

```diff
@@ -66,147 +65,177 @@ public void init(IWorkbench workbench)
+                }
 
-        return composite;
-    }
+                @Override
+                protected Object getValue(Object element) {
+                    if (element instanceof ConfirmationWithStatus) {
+                        return ((ConfirmationWithStatus) element).enabled;
+                    }
+                    return false;
+                }
 
-    private void createConfirmItem(@NotNull ConfirmationDescriptor descriptor) {
-        final TableItem item = new TableItem(confirmTable, SWT.NONE);
-        item.setData("id", descriptor.getId());
-        item.setText(0, descriptor.getTitle());
-        item.setText(1, descriptor.getGroup());
-        item.setText(2, getCurrentConfirmValue(descriptor.getId()));
-    }
+                @Override
+                protected void setValue(Object element, Object value) {
+                    if (element instanceof ConfirmationWithStatus) {
+                        ConfirmationWithStatus confirmation = (ConfirmationWithStatus) element;
+                        confirmation.enabled = CommonUtils.getBoolean(value, true);
+                        changedConfirmations.put((confirmation).confirmation, value);
+                    }
 
-    private void createConfirmItem(String group, ResourceBundle bundle, String id)
-    {
-        String labelKey = ConfirmationDialog.getResourceKey(id, ConfirmationDialog.RES_KEY_TITLE);
-        String title = bundle.getString(labelKey);
+                }
+            });
+
+        columnsController.addColumn(
+            CoreMessages.pref_page_confirmations_table_column_group,
+            CoreMessages.pref_page_confirmations_table_column_group,
+            SWT.RIGHT,
+            true,
+            true,
+            new ColumnLabelProvider() {
+            @Override
+            public String getText(Object element) {
+                if (element instanceof ConfirmationWithStatus) {
+                    return ((ConfirmationWithStatus) element).confirmation.getGroup();
+                }
+                return super.getText(element);
+            }
+        });
+
+        columnsController.createColumns(false);
+        tableViewer.setContentProvider(new ListContentProvider());
+        new DefaultViewerToolTipSupport(tableViewer);
+
+        Collection<ConfirmationDescriptor> descriptors = ConfirmationRegistry.getInstance().getConfirmations().stream()
+            // We do not want to see confirmation without a toggle message in the preferences
+            // because we do not want to add the user's ability to ignore these confirmations
+            .filter(item -> CommonUtils.isNotEmpty(item.getToggleMessage()))
+            .sorted(Comparator.comparing(ConfirmationDescriptor::getGroup))
+            .collect(Collectors.toList());
+        for (ConfirmationDescriptor confirmation : descriptors) {
+            this.confirmations.add(new ConfirmationWithStatus(confirmation, getCurrentConfirmValue(confirmation.getId())));
+        }

```

Comment:
And this can be pushed even further by mapping descriptors into `ConfirmationWithStatus`es in the `Stream` chain.

This way, we can improve the performance by a tiny bit and remove visual clutter.

```suggestion
       final var confirmations = ConfirmationRegistry.getInstance().getConfirmations().stream()
            // We do not want to see confirmation without a toggle message in the preferences
            // because we do not want to add the user's ability to ignore these confirmations
            .filter(item -> CommonUtils.isNotEmpty(item.getToggleMessage()))
            .sorted(Comparator.comparing(ConfirmationDescriptor::getGroup))
            .map(item -> new ConfirmationWithStatus(item, getCurrentConfirmValue(item.getId())))
            .collect(Collectors.toList());
      this.confirmations.addAll(confirmations);
```